sdk/ai/azure-ai-inference/src/main/java/com/azure/ai/inference/ChatCompletionClientTracer.java

```diff
@@ -0,0 +1,606 @@
+         * @return chat completions streaming for the provided chat messages.
+         */
+        Flux<StreamingChatCompletionsUpdate> invoke(BinaryData completeRequest, RequestOptions requestOptions);
+    }
+    //</editor-fold>
+
+    /**
+     * Creates ChatCompletionClientTracer.
+     *
+     * @param endpoint the service endpoint.
+     */
+    ChatCompletionClientTracer(String endpoint) {
+        this(endpoint, isContentCapturingEnabled(), null);
+    }
+
+    /**
+     * Creates ChatCompletionClientTracer.
+     *
+     * @param endpoint the service endpoint.
+     * @param captureContent if full content should be captured.
+     * @param clientOptions the client options.
+     */
+    ChatCompletionClientTracer(String endpoint, boolean captureContent, ClientOptions clientOptions) {
+        this.logger = new ClientLogger(ChatCompletionClientTracer.class);
+        final URL url = parse(endpoint, logger);
+        if (url != null) {
+            this.host = url.getHost();
+            this.port = url.getPort() == -1 ? url.getDefaultPort() : url.getPort();
+        } else {
+            this.host = null;
+            this.port = -1;
+        }
+        this.captureContent = captureContent;
+        this.tracer = createTracer(clientOptions);
+    }
+
+    /**
+     * Traces the synchronous convenience API - {@link com.azure.ai.inference.ChatCompletionsClient#complete(ChatCompletionsOptions)}.
+     *
+     * @param request input options containing chat options for complete API.
+     * @param operation the operation performing the actual completion call.
+     * @param completeRequest The completeRequest parameter for the {@code operation}.
+     * @param requestOptions The requestOptions parameter for the {@code operation}.
+     * @return chat completions for the provided chat messages.
+     */
+    @SuppressWarnings("try")
+    ChatCompletions traceSyncComplete(ChatCompletionsOptions request, SyncCompleteOperation operation,
+        BinaryData completeRequest, RequestOptions requestOptions) {
+        if (!tracer.isEnabled()) {
+            return operation.invoke(completeRequest, requestOptions);
+        }
+        final Context span = tracer.start(rootSpanName(request), new StartSpanOptions(SpanKind.CLIENT), Context.NONE);
+        traceCompletionRequestAttributes(request, span);
+        traceCompletionRequestEvents(request.getMessages(), span);
+
+        try (AutoCloseable ignored = tracer.makeSpanCurrent(span)) {
+            final ChatCompletions response = operation.invoke(completeRequest, requestOptions.setContext(span));
+            traceCompletionResponseAttributes(response, span);
+            traceCompletionResponseEvents(response, span);
+            tracer.end(null, null, span);
+            return response;
+        } catch (Exception e) {
+            tracer.end(null, e, span);
+            throw logger.logExceptionAsError(asRuntimeException(e));
```

Comment:
I don't think we can modify the exception - telemetry code should be transparent and not cause any changes to the underlying logic - we have to rethrow the same instance.